addons/netfox.extras/state-machine/network-state-machine.gd

```diff
@@ -0,0 +1,51 @@
+@icon("res://addons/netfox/icons/state-synchronizer.svg")
+extends Node
+class_name NetworkedStateMachine
+
+## A networked state machine that can be used to synchronize state between peers.
+
+@export var state: StringName = "": set = set_state
+
+signal state_changed(old_state: NetworkedState, new_state: NetworkedState)
+
+# Semi-private state property
+var _state_object: NetworkedState = null
+var _available_states: Dictionary = {}
+
+# Use the ready function to gather all available states
+func _ready():
+	for child in get_children():
```

Comment:
iirc you can use `find_children` to search by type, and you can disable the recursive search